### PR TITLE
docs: Add a guide for M3 theming

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/material-experimental/theming/m3.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# What is Material 3
+[Material 3 (M3)](https://m3.material.io/) is the latest iteration of Google's open-source design
+system, Material Design. It is the successor to [Material 2 (M2)](https://m2.material.io/), the
+design system to which Angular Material has historically adhered.
```
[the remark this answers is not in the document]
Comment (mmalerba) [reply src/material-experimental/theming/m3.md:6]: I think it sounds a little confusing with the 2 different senses of the word follows "it follows M2, which AM follows". How about we leave the "successor to" language for the first part?

File: src/material-experimental/theming/m3.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+### Customizing your colors
+The following aspects of your app's colors can be customized (see the
+[M3 color spec](https://m3.material.io/styles/color/roles) to learn more about these terms):
+
+- `theme-type`: Specifies the type of theme, `light` or `dark`.
+- `primary`: Specifies the palette to use for the app's primary color palette. (Note: the secondary,
+  neutral, and neutral-variant palettes described in the M3 spec will be automatically chosen based
+  on your primary palette, to ensure a harmonious color combination).
+- `tertiary`: Specifies the palette to use for the app's tertiary color palette.
```
[the remark this answers is not in the document]
Comment (mmalerba) [reply src/material-experimental/theming/m3.md:50]: They are actually all optional, but I can call that out explicitly

File: src/material-experimental/theming/m3.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+- `$m3-green-palette`
+- `$m3-blue-palette`
+- `$m3-yellow-palette`
+- `$m3-cyan-palette`
+- `$m3-magenta-palette`
+- `$m3-orange-palette`
+- `$m3-chartreuse-palette`
+- `$m3-azure-palette`
+- `$m3-violet-palette`
+- `$m3-rose-palette`
```
[the remark this answers is not in the document]
Comment (mmalerba) [reply src/material-experimental/theming/m3.md:64]: Yeah, I want to go through and add a few embedded examples to illustrate some of these things, but I'll do it in a followup

**Review by andrewseguin — APPROVED** [review]

File: guides/material-3.md [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+    primary: matx.$m3-violet-palette,
+  ),
+  typography: (
+    brand: 'Comic Sans',
+    bold: 900
+  ),
+  density: (
+    scale: -1
+  )
+));
```
[the remark this answers is not in the document]
Comment (andrewseguin) [reply guides/material-3.md:45]: or is it..

```
html {
  @include mat.button-theme($theme);
}
```

**Pull request merged**